common/predictive-text/worker/model-compositor.ts

```diff
@@ -145,4 +154,38 @@ class ModelCompositor {
 
     return suggestions;
   }
-}
\ No newline at end of file
+
+  /**
+   * Returns the punctuation used for this model, filling out unspecified fields
+   */
+  private static determinePunctuationFromModel(model: WorkerInternalModel): LexicalModelPunctuation {
```

Comment:
Couldn't we just use `Object.assign({}, DEFAULT_PUNCTUATION, model.punctuation)` for this function? That should even handle the `''` case AFAICT.

Reply:
There's a few reasons I didn't do this. `Object.assign()` does not recurse into objects, e.g.,

```javascript
var defaults = {
  quotes: { open: "“", close: "”" },
  afterWord: " ",
};

var customizations = {
  afterWord: "",
  quotes: { open: "„" }  // only override one quote
};

var c = Object.assign({}, defaults, customizations);
```


Expected:
```javascript
c == { quotes: { open: '„', close: "”" }, afterWord: '' }
```

Actual:
```javascript
c == { quotes: { open: '„' }, afterWord: '' }
```

A second reason is that `Object.assign()` is not available in IE 11: [[1]](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Browser_compatibility) [[2]](https://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign_a_href=_https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign_title=_MDN_documentation_img_src=_../mdn.png_alt=_MDN_(Mozilla_Development_Network)_logo_width=_15_height=_13_/_/a_nbsp;)